config/config_encryption_test.go

```diff
@@ -356,11 +356,10 @@ func TestSaveConfigToFileWithErrorInPasswordPrompt(t *testing.T) {
 		EncryptConfig: fileEncryptionEnabled,
 	}
 	testData := []byte("testdata")
-	f, err := os.CreateTemp("", "")
+	f, err := os.CreateTemp(t.TempDir(), "")
```

Comment:
I think this is a thing already brought up in other comments.You don't need to create the dir.
From `TempDir()` comments:
> // if the directory creation fails, TempDir terminates the test by calling Fatal.

Implying creation occurs

Reply:
That comment was in relation to our previous usage of `defer os.Remove(f.Name())`, that wasn't needed since I switched us over to using `t.TempDir()` so that does it for us. I removed all the ones I could find but one was missing. `os.CreateTemp` actually creates a temporary file (not a directory), so this is still needed as t.TempDir returns a directory which still gets auto-deleted once the test finishes